realm/realm-library/src/main/cpp/io_realm_internal_Table.cpp

```diff
@@ -20,8 +20,18 @@
 #include "io_realm_internal_Table.h"
 #include "tablebase_tpl.hpp"
 
+#include "util/format.hpp"
+
+#include "jni_util/java_exception_thrower.hpp"
+#include "java_exception_def.hpp"
+
 using namespace std;
 using namespace realm;
+using namespace realm::_impl;
+using namespace realm::jni_util;
+using namespace realm::util;
+
+static const char* c_null_values_cannot_set_required_msg = "The primary key field '%1' has 'null' values stored. It cannot be converted to required primary key field.";
```

Comment:
It cannot be converted to required primary key field. -> It cannot be converted to a @Required primary key field.